src/pint/models/stand_alone_psr_binaries/binary_orbits.py

```diff
@@ -237,3 +237,280 @@ def d_pbprime_d_par(self, par):
             if re.match(r"FB\d+", par) is not None
             else np.zeros(len(self.tt0)) * u.second / par_obj.unit
         )
+
+
+class OrbitWaves(Orbit):
+    """Orbit with orbital phase variations described by a Fourier series"""
+
+    def __init__(
+        self,
+        parent,
+        orbit_params=[
+            "PB",
+            "TASC",
+            "ORBWAVE_OM",
+            "ORBWAVE_EPOCH",
+            "ORBWAVEC0",
+            "ORBWAVES0",
+        ],
```

Comment:
Wouldn't it be better to have PBDOT also here to capture a linear secular variation of PB? That way ORBWAVE_OM can be the same as 1/Tspan like we have in red noise.

Reply:
Indeed, @kerrm has added `OrbitWavesFBX` in that file for that purpose. I'll add a test case for that model too.